kobo/apps/accounts/mfa/tests/test_api.py

```diff
@@ -67,3 +73,36 @@ def test_mfa_activation_always_creates_new_secret(self):
             )
             assert first_secret != second_secret
             assert first_response.json() != second_response.json()
+        mfa_availability.delete()
+
+    @override_config(MFA_ENABLED=True)
+    def test_mfa_disabled(self):
+        # Enable the MFA whitelist by adding a user
+        someuser_mfa_activation = MfaAvailableToUser.objects.create(
+            user=self.someuser
+        )
+
+        anotheruser = User.objects.get(username='anotheruser')
+        self.client.login(username='anotheruser', password='anotheruser')
+        method = list(trench_settings.MFA_METHODS.keys())[0]
+
+        activate_response = self.client.post(
+            reverse('mfa-activate', args=(method,))
+        )
+        assert activate_response.status_code == status.HTTP_403_FORBIDDEN
+
+        mfa_availability = MfaAvailableToUser.objects.create(user=anotheruser)
+        activate_response = self.client.post(
+            reverse('mfa-activate', args=(method,))
+        )
+        assert activate_response.status_code == status.HTTP_200_OK
+
+        # Reset MFA whitelist state
+        mfa_availability.delete()
+        someuser_mfa_activation.delete()
+
+        # Test when whitelist is disabled
```

Comment:
Would it be better to test if the whitelist is really empty before testing the activation?

Reply:
Yes, moved it